src/swap/changehero.ts

```diff
@@ -134,11 +161,26 @@ export function makeChangeHeroPlugin(
       getAddress(request.fromWallet),
       getAddress(request.toWallet)
     ])
-    const { fromCurrencyCode, toCurrencyCode } = getCodes(request)
 
-    // The chain codes are undocumented but ChangeHero uses Edge pluginIds for these values (confirmed via Slack)
-    const fromMainnetCode = request.fromWallet.currencyInfo.pluginId
-    const toMainnetCode = request.toWallet.currencyInfo.pluginId
+    // Supported chains must be whitelisted
+    if (
+      MAINNET_CODE_TRANSCRIPTION[request.fromWallet.currencyInfo.pluginId] ==
+        null ||
+      MAINNET_CODE_TRANSCRIPTION[request.toWallet.currencyInfo.pluginId] == null
+    ) {
+      throw new SwapCurrencyError(
+        swapInfo,
+        request.fromCurrencyCode,
+        request.toCurrencyCode
```

Comment:
`throw new SwapCurrencyError(swapInfo, request)`. Passing currency codes is deprecated.

Reply:
I forgot about that. Better fix them all!